src/mediapipe/imageSegmenter/audioNativeRenderer/MatrixAudioRenderer.tsx

```diff
@@ -0,0 +1,144 @@
+   * @alpha
+   */
+  muted?: boolean;
+  /**
+   * The list of participants to render audio for.
+   * This list needs to be composed based on the matrixRTC members so that we do not play audio from users
+   * that are not expected to be in the rtc session.
+   */
+  members?: CallMembership[];
+  /**
+   * The balance of the audio tracks.
+   * This is a value between -1 and 1, where -1 is full left, 0 is center, and 1 is full right.
+   */
+  balance?: number;
+}
+
+/**
+ * The `RoomAudioRenderer` component is a drop-in solution for adding audio to your LiveKit app.
+ * It takes care of handling remote participants’ audio tracks and makes sure that microphones and screen share are audible.
+ *
+ * @example
+ * ```tsx
+ * <LiveKitRoom>
+ *   <NativeOrWebviewMatrixAudioRenderer />
+ * </LiveKitRoom>
+ * ```
+ * @public
+ */
+export function MatrixAudioRenderer({
+  volumeFactor,
+  muted,
+  members,
+  balance,
+}: RoomAudioRendererProps): ReactNode {
+  const validIdentities = useMemo(
+    () => members?.map((member) => member.sender + ":" + member.deviceId),
+    [members],
+  );
+
+  const tracks = useTracks(
+    [
+      Track.Source.Microphone,
+      Track.Source.ScreenShareAudio,
+      Track.Source.Unknown,
+    ],
+    {
+      updateOnlyOn: [],
+      onlySubscribed: true,
+    },
+  ).filter((ref) => {
+    const isRemote = !ref.participant.isLocal;
+    const isAudio = ref.publication.kind === Track.Kind.Audio;
+    const isValidMatrixMember = validIdentities?.includes(
+      ref.participant.identity,
+    );
+    if (!isValidMatrixMember) {
+      logger.warn(
+        `Audio track ${ref.participant.identity} has no matching matrix call member`,
+        `current members: ${validIdentities}`,
+        `track will not get rendered`,
+      );
+    }
+    return isRemote && isAudio && isValidMatrixMember;
+  });
+
+  const audioContext = useRef<AudioContext>(new AudioContext());
+  const trackNodes = useRef<
+    Map<string, { pan: StereoPannerNode; gain: GainNode }>
+  >(new Map());
+  useEffect(() => {
+    tracks.forEach((trackRef) => {
+      const track = trackRef.publication.track as RemoteAudioTrack;
+      const nodes = trackNodes.current.get(trackRef.participant.identity);
+      const ctx = audioContext.current;
+      if (nodes) {
+        const { pan, gain } = nodes;
+        pan.pan.value = balance ?? 0;
+        gain.gain.value = volumeFactor ?? 1;
+      } else {
+        track.setAudioContext(ctx);
+
+        const panNode = ctx.createStereoPanner();
+        panNode.pan.value = balance ?? 0;
+        const gainNode = ctx.createGain();
+        gainNode.gain.value = volumeFactor ?? 1;
+        track.setWebAudioPlugins([panNode, gainNode]);
+        trackNodes.current.set(trackRef.participant.identity, {
+          pan: panNode,
+          gain: gainNode,
+        });
+      }
+
+      // Remove nodes from trackNodes map.
+      trackNodes.current.forEach((_, identity) => {
```

Comment:
Do we need to do any cleanup of the audioContext when removing tracks? Can they end up added twice or something?